services-core/src/main/java/com/mapbox/core/utils/TextUtils.java

```diff
@@ -111,27 +126,44 @@ public static String formatRadiuses(double[] radiuses) {
     return join(";", radiusesFormatted);
   }
 
+  public static String formatRadiuses(List<Double> radiuses) {
+    if (radiuses == null || radiuses.size() == 0) {
+      return null;
+    }
+
+    String[] radiusesFormatted = new String[radiuses.size()];
+    for (int i = 0; i < radiuses.size(); i++) {
+      if (radiuses.get(i) == Double.POSITIVE_INFINITY) {
+        radiusesFormatted[i] = "unlimited";
+      } else {
+        radiusesFormatted[i] = String.format(Locale.US, "%s",
+                TextUtils.formatCoordinate(radiuses.get(i)));
+      }
+    }
+    return join(";", radiusesFormatted);
+  }
+
   /**
    * Formats the bearing variables from the raw values to a string which can than be used for the
    * request URL.
    *
-   * @param bearings a List of doubles representing bearing values
+   * @param bearings a List of list of doubles representing bearing values
    * @return a string with the bearing values
    * @since 3.0.0
    */
-  public static String formatBearing(List<Double[]> bearings) {
+  public static String formatBearing(List<List<Double>> bearings) {
```

Comment:
might `List<Double>` changed to `Pair` or custom model?

Reply:
yep, agree, I just wanted to make it in a separate PR to make review and testing easier

Reply:
leave `TODO` to not forget 😃